.github/workflows/integration-test.yml

```
@@ -0,0 +1,44 @@
+name: Integration Test
+
+env:
+  TRAVIS: 1
+
+on: [pull_request]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout project
+      uses: actions/checkout@v6
+
+    - name: Specify node version
+      uses: actions/setup-node@v6
+      with:
+        node-version: 22
+        cache: 'npm'
+        cache-dependency-path: package-lock.json
+
+    - name: Cache node modules
+      id: cache-node-modules
+      uses: actions/cache@v4
+      with:
+          path: node_modules
+          key: node-modules-${{ hashFiles('package-lock.json') }}
+    - name: Install Dependencies
+      if: steps.cache-node-modules.outputs.cache-hit != 'true'
+      run: npm install
```

npm ci is preferred in CI environments as it's faster, more reliable, and ensures reproducibility by strictly following package-lock.json.

```suggestion
      run: npm ci
```